### PR TITLE
fix: updating message for po payments

### PR DESCRIPTION
**Pull request by deov31**
Jira: [BUN-1641](https://bigc-b2b.atlassian.net/browse/BUN-1641)

## What/Why?
I'm changing the message of the payments details when seeing any order detail because is confusing for buyers and sellers who still owe payment via Invoice

## Rollout/Rollback
Revert PR 

## Testing
image

### CODE REVIEW

File: apps/storefront/src/pages/OrderDetail/components/OrderAction.tsx [hunk 1]
```diff
@@ -285,8 +285,19 @@ export default function OrderAction(props: OrderActionProps) {
...
+  const getPaymentMessage = () => {
+    const message = poNumber ? 'orderDetail.paidWithPo' : 'orderDetail.paidInFull';
+    return (
+      createAt &&
+      b3Lang(message, {
+        paidDate: displayFormat(createAt, true),
+      })
+    );
+  };
+
```
Comment (bc-marco) [inline apps/storefront/src/pages/OrderDetail/components/OrderAction.tsx:300]: As you're using a prop you should use `useCallback`
```suggestion
  const getPaymentMessage = useCallback(() => {
    const message = poNumber ? 'orderDetail.paidWithPo' : 'orderDetail.paidInFull';
    let message = ''
    
    if(createdAt) {
      if(poNumber) {
        message = b3Lang('orderDetail.paidWithPo', {
          paidDate: displayFormat(createAt, true),
        })
      } else {
        message = b3Lang('orderDetail.paidInFull', {
          paidDate: displayFormat(createAt, true),
        })
      }
    }

    return message
  }, [poNumber, createAt]);

```

File: apps/storefront/src/pages/OrderDetail/components/OrderAction.tsx [hunk 2]
```diff
@@ -407,11 +418,7 @@ export default function OrderAction(props: OrderActionProps) {
     {
       header: b3Lang('orderDetail.payment'),
       key: 'payment',
-      subtitle: createAt
-        ? b3Lang('orderDetail.paidInFull', {
-            paidDate: displayFormat(createAt, true),
-          })
-        : '',
+      subtitle: getPaymentMessage() || '',
```
Comment (bc-marco) [inline apps/storefront/src/pages/OrderDetail/components/OrderAction.tsx:421]: With the suggestion above there is no need to validate here, just use what is returned from `getPaymentMessage()`
```suggestion
      subtitle: getPaymentMessage(),
```

**Review by bc-marco — CHANGES_REQUESTED** [review]
Overall looks good, there are suggestions to improve readability

Feel free to ask any questions

File: apps/storefront/src/pages/OrderDetail/components/OrderAction.tsx (hunk 1 above)
Comment (deov31) [reply apps/storefront/src/pages/OrderDetail/components/OrderAction.tsx:300]: 👍

**Review by bc-marco — APPROVED** [review]

**Review by bc-victor — APPROVED** [review]

**Pull request merged**